packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -10,15 +10,26 @@
 // for more info.
 static NSString *const kClientIdKey = @"CLIENT_ID";
 
+// These error codes must match with ones declared on Android and Dart sides.
+static NSString *const kErrorReasonSignInRequired = @"sign_in_required";
+static NSString *const kErrorReasonSignInCanceled = @"sign_in_canceled";
+
 @interface NSError (FlutterError)
 @property(readonly, nonatomic) FlutterError *flutterError;
 @end
 
 @implementation NSError (FlutterError)
 - (FlutterError *)flutterError {
-  return [FlutterError errorWithCode:@"exception"
-                             message:self.domain
-                             details:self.localizedDescription];
+  NSString *errorCode;
+  if (self.code == kGIDSignInErrorCodeHasNoAuthInKeychain) {
+    errorCode = kErrorReasonSignInRequired;
+  } else if (self.code == kGIDSignInErrorCodeCanceled) {
+    errorCode = kErrorReasonSignInCanceled;
+  } else {
+    errorCode = @"exception";
```

Comment:
This is `status` on the Android side, and no exception was actually thrown in ObjC code. I think we should make this catch-all error code consistent as well.